### PR TITLE
MLIBZ-1317 Make forgotUsername(), verifyEmail(), and resetPassword() functions on the User class static

### PR DESCRIPTION
**Pull request by ThomasConner**
#### Description
`forgotUsername()`, `verifyEmail()`, and `resetPassword()` functions on the `User` class should be static and not require an active user to be executed.

#### Changes
- Made `forgotUsername()`, `verifyEmail()`, and `resetPassword()` functions on the `User` class.
- Added unit tests for the changes made on the `User` class.

### CODE REVIEW

File: src/entity/src/user.js [hunk 1]
```diff
@@ -785,18 +807,40 @@ export class User {
    * @return {Promise<Object>} The response.
    */
   async forgotUsername(options = {}) {
```
Comment (tejasranade) [inline src/entity/src/user.js]: @ThomasConner  In the interest of avoiding confusion, should we deprecate this instance method? Seems like it's meaningless anyway to call `forgotUsername` when you already have a user instance. I have similar comments about a couple of other user verification APIs. We could start by adding deprecation notes and eventually taking them out. Thoughts?

Comment (ThomasConner) [inline src/entity/src/user.js]: Yea I like that idea. Would you like me to add deprecation documentation for this release?

Comment (tejasranade) [inline src/entity/src/user.js]: Yes, please. Please also verify that the docs refer to the correct API.

Comment (ThomasConner) [inline src/entity/src/user.js]: Done

**Pull request merged**